File: training task/encapsulation_private.py
class StudentPrivate:
    def __init__(self, name, age):
        self.__name = name 
        self.__age = age    

    def get_name(self):
        return self.__name

   
    def set_name(self, name):
        self.__name = name

  
    def get_age(self):
        return self.__age


    def set_age(self, age):
        self.__age = age

    def display_details(self):
        print(f"Name: {self.__name}, Age: {self.__age}")


student = StudentPrivate("Mohua", 20)

print("Original Details:")
student.display_details()

student.set_name("Mukta")
student.set_age(22)

print("Modified Details:")
student.display_details()

File: training task/encapsulation_private.cpp
#include <iostream>
#include <string>

using namespace std;

class StudentPrivate {
private:
    string name;
    int age;

public:
    StudentPrivate(string name, int age){
         this->name = name;
    this->age = age;
    }   
    string getName() ;
    void setName(string name);
    int getAge();
    void setAge(int age);
    void displayDetails() ;
};



string StudentPrivate::getName(){
    return name;
}


void StudentPrivate::setName(string name) {
    this->name = name;
}


int StudentPrivate::getAge(){
    return age;
}


void StudentPrivate::setAge(int age) {
    this->age = age;
}


void StudentPrivate::displayDetails() {
    cout << "Name: " << name << ", Age: " << age << endl;
}

int main() {
    StudentPrivate student("Sadia", 20);

    cout << "Original Details:" << endl;
    student.displayDetails();

    student.setName("Mohua");
    student.setAge(22);

    cout << "Modified Details:" << endl;
    student.displayDetails();

    return 0;
}
